### ISSUE TITLE
[toc] not rendered if first tocItem is not in view

### ISSUE THREAD

**Issue by dcheglakov** [vnphanquang/svelte-put#133]
Hi,

I've an issue with `svelte-put/toc`. Table of contents is not rendered if first `tocItem` is not in view on page load.
In my case I'm targeting for h2 elements and first h2 is not in view.

Here is [svelte repl](https://svelte.dev/repl/5f61ba3ad7664901bc899aec75d6b658?version=3.55.1) with this issue.

**Comment by vnphanquang** [comment]
@dcheglakov hello there, I see there are 2 headings in your REPL:

- one is `<h2>Title 1</h2>` and is extracted by `toc` sucessfully,
- the other one is `<h2 data-toc-ignore>Contents</h2>` which is ignored by `toc` because of the `data-toc-ignore` attribute added there.

If you are referring to `<h2 data-toc-ignore>Contents</h2>`, then just remove the `data-toc-ignore`. Does that help?

**Comment by dcheglakov** [comment]
@vnphanquang in my case `<h2>Title 1</h2>` is **not** extracted by `toc`. When you tested [svelte repl](https://svelte.dev/repl/5f61ba3ad7664901bc899aec75d6b658?version=3.55.1) is `<h2>Title 1</h2>` was in the view or outside of it?

Here is what it looks like when `<h2>Title 1</h2>` is not in view on page load

<img width="1300" alt="image" src="https://user-images.githubusercontent.com/7033855/217743246-cb2613bd-2a08-48b0-a342-70db6e7da66f.png">

Notice the `table of contents` is **empty**

And after scrolling `<h2>Title 1</h2>` is appered in `toc`
<img width="1300" alt="image" src="https://user-images.githubusercontent.com/7033855/217743593-b7abf698-1893-43e6-a517-dfb6a8ac53ea.png">

**Comment by vnphanquang** [comment]
@dcheglakov hmm interesting. From my side everything is okay, tested in chrome and firefox. Here is a loom: https://www.loom.com/share/3c057f54bbe64992bca30f77b8075f07

Can you provide some more info about your env (browser, os,...). Also does this occur in your project on localhost or just in REPL sandbox? Let me know

**Comment by dcheglakov** [comment]
@vnphanquang the problem occurs when `<h2>Title 1</h2>` is not in view. Just copy paste paragraphs to shift `<h2>Title 1</h2>` outside the view. Or use this [updated repl ](https://svelte.dev/repl/5f61ba3ad7664901bc899aec75d6b658?version=3.55.1)with more `p` tags.

And yes I've this problem in the real project I'm currently working on.

**Comment by vnphanquang** [comment]
@dcheglakov Ah i see, apologies I wasn't understanding what you meant by 'inside/outside of view'. Taking a look right now

**Comment by vnphanquang** [comment]
@dcheglakov should be resolved with [@svelte-put/toc@2.1.3](https://github.com/vnphanquang/svelte-put/releases/tag/%40svelte-put/toc%402.1.3) (I forgot to run it on moun, silly me). 

You can refresh your REPL and verify the fix. Thanks for raising this and for your patience.

**Comment by dcheglakov** [comment]
@vnphanquang thank you for such a quick fix. Everything is fine now. Thanks again.

**Comment by vnphanquang** [comment]
@dcheglakov no worries, happy that folks are using this.

Just one thing, I notice you've placed `data-toc-observe-for` in your `div`, this won't have any effect because `data-toc-observe-for` is `readonly` and meant for internal references. The docs wasn't clear on this but I've updated it recently. Hope that helps:

https://svelte-put.vnphanquang.com/docs/toc#by-observe-operation

**Comment by dcheglakov** [comment]
@vnphanquang yeah forgot to remove it. I was desperate and tried everything possible and imposible to fix it 😂
